tests_in_tilt/Tiltfile

```diff
@@ -0,0 +1,51 @@
+# Some common test configurations to easily get up and running with Tilt for Tests
+# (see docs: https://docs.tilt.dev/tests_in_tilt.html)
+
+### GOLANG TESTS ###
+def test_go_all(path, deps, timeout='', tags=None, mod=''):
```

Comment:
1. The first arg is a package, not a path (e.g., `'./internal/tiltfile'` and `'github.com/tilt-dev/tilt/internal/tiltfile'` are valid, but `'internal/tiltfile'` is not). This is also what go test -help calls its arg. I'm not sure if we're adding much value by forcing it to be recursive if we're already asking the user to know how to specify a go package.
2. nit: I think `timeout` and `mod` should default to `None` or `tags` should default to `[]`, to be consistent.

Reply:
1. that makes sense, i changed the arg name and added a `recursive` bool arg in case they want us to add the `/...` for them
2. alas, the [mutable default args](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments) gotcha is still a thing in Starlark :( so i think there's a strong argument to be made for "default value of the type, if possible; `None` where it's dangerous". This is the way I've seen it done in lots of other python code fwiw.

Reply:
> alas, the mutable default args gotcha is still a thing in Starlark :(

computers gonna compute :(